pkg/instance/service.go

```diff
@@ -168,23 +171,26 @@ func (s service) Logs(instance *model.Instance, group *models.Group) (io.ReadClo
 	return read, err
 }
 
-func (s service) getPod(client *kubernetes.Clientset, instance *model.Instance) v1.Pod {
+func (s service) getPod(client *kubernetes.Clientset, instance *model.Instance, selector string) (v1.Pod, error) {
```

Comment:
Can you add a guard (returning an error) here if selector is anything other than `""` or `"data"` like we talked about. So we are reminded of the connection to stacks (don't accidentally allow streaming of `im-data-secret-id`, ...).